gitlab-ci.yml

```diff
@@ -0,0 +1,42 @@
+cache:
+  key:
+    files:
+      # Cache npm packages using lock file
+      - package-lock.json
+  paths:
+    - .npm/
+
+before_script:
+    - npm ci --cache .npm --legacy-peer-deps --loglevel error
+
+stages:
+  - lint
+  - test
+
+reviewdog:
+  image: node:16.16
+  stage: lint
+  script:
+    - mkdir -p ~/bin/ && export PATH="~/bin/:$PATH"
+    - curl -sfL https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh| sh -s -- -b ~/bin
+    - export REVIEWDOG_GITLAB_API_TOKEN=$CODE_REVIEW_SHIFU_KEY
+    - reviewdog -reporter=gitlab-mr-discussion
+
+unit_test:
+  image: node:16.16
+  stage: test
+  script:
+    - mkdir -p ~/bin/ && export PATH="~/bin/:$PATH"
+    - npm run jest:ci
+
+e2e_test:
+  image: cypress/browsers:node16.16.0-chrome107-ff107
```

Comment:
same as above. Let's update node version to 18 as we need to maintain consistency through out the application. Also, Let's make sure we have update chrome version too.